pkg/api/api.go

```diff
@@ -239,7 +239,10 @@ func (a *API) RegisterRuntimeConfig(runtimeConfigHandler http.HandlerFunc) {
 func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config) {
 	distributorpb.RegisterDistributorServer(a.server.GRPC, d)
 
-	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, a.cfg.wrapDistributorPush(d)), true, false, "POST")
+	wrappedDistributor := a.cfg.wrapDistributorPush(d)
+
+	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, wrappedDistributor), true, false, "POST")
+	a.RegisterRoute("/api/v1/push/otlp/v1/metrics", push.HandlerForOTLP(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, wrappedDistributor), true, false, "POST")
```

Comment:
Why is this under `/api/v1/push`? 
It looks weird to me, because it's a non-prometheus api under a prometheus API path. I'd expect it to be `/api/oltp/v1/metrics/push` or similar instead. 

Is it based on something already existing in other projects?